brontide/conn.go

```diff
@@ -234,7 +234,10 @@ func (c *Conn) Flush() (int, error) {
 //
 // Part of the net.Conn interface.
 func (c *Conn) Close() error {
-	// TODO(roasbeef): reset brontide state?
+	// Clear the state we created to be able to handle this connection.
+	c.noise = nil
+	c.readBuf = bytes.Buffer{}
+
```

Comment:
This is a null deref waiting to happen.  It is common for Brontide goroutines to continue calling `Conn` methods after the connection has been closed.

Reply:
Currently [disconnect is guarded by an atomic variable](https://github.com/Roasbeef/lnd/blob/483dc3f96a73f9e719ba92e8a5bc5697f50b9c80/peer.go#L424-L440), so callers to it will only succeed once. 

If I look at the other callers into `p.conn`, I see one one [the string method](https://github.com/Roasbeef/lnd/blob/483dc3f96a73f9e719ba92e8a5bc5697f50b9c80/peer.go#L442-L445). But not many others. 

In either case, we can have a similar CAS check on calls into the connection after a it's been closed. Thanks for pointing this out.